openmc/statepoint.py

```diff
@@ -94,475 +91,385 @@ def __init__(self, filename):
+                    if filter_type in ['energy', 'energyout']:
+                        bins = self._f['{0}{1}/bins'.format(subbase, j)].value
 
-        # Check whether Tally results have been read
-        if not self._results:
-            self.read_results()
+                    elif filter_type in ['mesh', 'distribcell']:
+                        bins = self._f['{0}{1}/bins'.format(subbase, j)].value
 
-        # Check if source bank is in statepoint
-        if not self._source_present:
-            print('Unable to read source since it is not in statepoint file')
-            return
+                    else:
+                        bins = self._f['{0}{1}/bins'.format(subbase, j)].value
 
-        # Initialize a NumPy array for the source sites
-        self._source = np.empty(self._n_particles, dtype=SourceSite)
+                    # Create Filter object
+                    filter = openmc.Filter(filter_type, bins)
+                    filter.offset = offset
+                    filter.num_bins = n_bins
 
-        # For HDF5 state points, copy entire bank
-        source_sites = self._f['source_bank'].value
+                    if filter_type == 'mesh':
+                        mesh_ids = self._f['tallies/meshes/ids'].value
+                        mesh_keys = self._f['tallies/meshes/keys'].value
 
-        # Initialize SourceSite object for each particle
-        for i in range(self._n_particles):
-            # Initialize new source site
-            site = SourceSite()
+                        key = mesh_keys[mesh_ids == bins][0]
+                        filter.mesh = self.meshes[key]
 
-            # Read position, angle, and energy
-            site._weight, site._xyz, site._uvw, site._E = source_sites[i]
+                    # Add Filter to the Tally
+                    tally.add_filter(filter)
 
-            # Store the source site in the NumPy array
-            self._source[i] = site
+                # Read Nuclide bins
+                nuclide_names = self._f['{0}{1}/nuclides'.format(base, tally_key)].value
 
-    def compute_ci(self, confidence=0.95):
-        """Computes confidence intervals for each Tally bin.
+                # Add all Nuclides to the Tally
+                for name in nuclide_names:
+                    nuclide = openmc.Nuclide(name.decode().strip())
+                    tally.add_nuclide(nuclide)
 
-        This method is equivalent to calling compute_stdev(...) when the
-        confidence is known as opposed to its corresponding t value.
+                # Read score bins
+                n_score_bins = self._f['{0}{1}/n_score_bins'.format(base, tally_key)].value
 
-        Parameters
-        ----------
-        confidence : float, optional
-            Confidence level. Defaults to 0.95.
+                tally.num_score_bins = n_score_bins
 
-        """
+                scores = self._f['{0}{1}/score_bins'.format(
+                    base, tally_key)].value
```

Comment:
I think that adding whitespaces to this line to make it line up with indexing bracket above is easier to read (as you did for ``n_user_scores`` below).